plugins/chat/assets/javascripts/discourse/services/chat-channels-manager.js

```diff
@@ -30,6 +32,20 @@ export default class ChatChannelsManager extends Service {
     }
   }
 
+  // TODO (martin) Maybe we should make a ChatThreadManager as well?
```

Comment:
I think you will agree with this, seems like too much mixing of concerns, maybe we should have one ChatThreadManager instance per channel in ChatChannelManager, thoughts? All the other functions I added in this file would be inclluded.